server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -573,6 +581,9 @@ public void writeTo(StreamOutput out) throws IOException {
                     out.writeList(Collections.emptyList());
                 }
             }
+            if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_062)) {
+                out.writeBoolean(ccsMinimizeRoundtrips);
```

Comment:
Technically this field does not need to be serialized for production use, but I found that many of the IT tests fail with wonky results if I don't propagate this, so I finally added it. I probably should have done that from the start.

Reply:
I'm not sure we should serialize a field just so that we can use it in tests. In fact, it appears as if we only use this in a single non-assertion production spot to begin with? That spot has the rewritten request in scope, so couldn't we just determine the `ccsMinimizeRoundTrips` value from the request in that spot and do away with tracking it here?
We already have a mutable map in this thing, would be nice if we could save ourselves the mental overhead of serialising redundant data here? (I get that it's just a boolean but we had some serious issues in the past with search responses containing redundant data, so I'd rather not restart that trend :D )

Reply:
I changed `ccsMinimizeRoundtrips` in `SearchResponse.Clusters` to no longer be serialized. I had to make `TransportSearchAction.shouldMinimizeRoundtrips` public so I can use them in IT tests. With that change the tests are passing.